### DISCUSSION TITLE
External grader docker image for Matlab

### DISCUSSION THREAD

**Discussion by Victorsss-Orz** [PrairieLearn/PrairieLearn#9230] [Show and tell]
I have been working on creating an autograder for Matlab for UIUC CS101, mimicking the structures used by the python autograder. Currently I have a version that can work on simple tasks such as running student code, retrieving or calling a student-defined variable or function, and passing variables (not functions yet) to the student code. The grader utilizes octave for executing Matlab code, and the source code and docker image are both public. Feel free to play with it! I'll continue to working on more functionalities. Any suggestions and ideas for improvements are much welcomed! 
https://github.com/Victorsss-Orz/matlab_autograder

**Comment by lslavice** [comment]
THANK YOU for this!! We're having trouble with our students using GenAI to write their code (who isn't these days) and it's coming up with all kinds of wacky MATLAB things. We have a separate autograder site that uses MATLAB itself, but limiting what will work via Octave will actually work better for us these days. Plus, we can use PL 🙂
